pagerduty/resource_pagerduty_incident_workflow.go

```diff
@@ -172,13 +343,7 @@ func resourcePagerDutyIncidentWorkflowCreate(ctx context.Context, d *schema.Reso
 		return diag.FromErr(err)
 	}
 
-	stepIdMapping := map[int]string{}
-	for i, s := range createdWorkflow.Steps {
-		stepIdMapping[i] = s.ID
-	}
-
-	nonGeneratedInputNames := createNonGeneratedInputNamesFromWorkflowStepsWithoutIDs(iw, stepIdMapping)
-	err = flattenIncidentWorkflow(d, createdWorkflow, true, nonGeneratedInputNames)
+	err = flattenIncidentWorkflow(d, createdWorkflow, true, specifiedSteps)
```

Comment:
When we flatten the workflow after receiving the response, we now use the specified inputs tracked earlier to determine which inputs need `generated=true` or not.